app/models/photo.rb

```diff
@@ -143,4 +156,39 @@ def self.visible(current_user, person, limit=:all, max_time=nil)
              end
     photos.where(pending: false).order("created_at DESC")
   end
+
+  def coords_to_float(deg, min, sec)
+    (deg.to_r + min.to_r / 60 + sec.to_r / 3600).to_f
+  end
+
+  def determine_location(image)
+    exif_gps_latitude  = image.exif["GPSLatitude"]
+    return nil unless exif_gps_latitude
+
+    exif_gps_longitude = image.exif["GPSLongitude"]
+    return nil unless exif_gps_longitude
+
+    lat = coords_to_float(*exif_gps_latitude.split(", ")) * (image.exif["GPSLatitudeRef"] == "S" ? -1 : 1)
+    lng = coords_to_float(*exif_gps_longitude.split(", ")) * (image.exif["GPSLongitudeRef"] == "W" ? -1 : 1)
+
+    nominatim_request =
+      Typhoeus.get("https://nominatim.openstreetmap.org/reverse?" +
```

Comment:
We are not allowed to use nominatim. Check their [usage policy](https://wiki.openstreetmap.org/wiki/Nominatim_usage_policy), highlights by me:

> an absolute maximum of 1 request per second [...]
>
> Provide a valid HTTP Referer or User-Agent identifying the application (**stock User-Agents as set by http libraries will not do**). [...] 
> 
> Clearly display attribution as suitable for your medium.
> 
> Use that is directly **triggered by the end-user** (for example, user searches for something) is ok, provided that your number of users is moderate. Note that the usage limits above apply per website/application: the sum of traffic by all your users should not exceed the limits.